src/checkers/inference/solver/backend/z3backend/Z3BitVectorCodec.java

```diff
@@ -0,0 +1,24 @@
+package checkers.inference.solver.backend.z3backend;
+
+import java.math.BigInteger;
+
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.AnnotationMirror;
+
+public interface Z3BitVectorCodec {
+
+    /**
+    * Get the fixed Bit Vector size.
+    * @return the fixed Bit Vector size
+    */
+    int getFixedBitVectorSize();
+
+    /**
+     * Encode a given AnnotationMirror to a numeral value represents the encoded bit vector.
```

Comment:
Grammar

Reply:
I'm not sure what the proper meaning is by the sentence, but this is my guess:

"Encode a given AnnotationMirror into a numeric value, for use in a encoded bit vector."

Reply:
We use a bigInteger to represent the encoded bit vector, i.e. the binary representation of this numeric value is the encoded bit vector.

I've updated the javadoc to make it a bit clear.